lib/net.js

```diff
@@ -444,23 +444,21 @@ Socket.prototype.destroySoon = function() {
 Socket.prototype._destroy = function(exception, cb) {
   debug('destroy');
 
-  var self = this;
-
   function fireErrorCallbacks() {
     if (cb) cb(exception);
-    if (exception && !self._writableState.errorEmitted) {
-      process.nextTick(emitErrorNT, self, exception);
-      self._writableState.errorEmitted = true;
+    if (exception && !this._writableState.errorEmitted) {
+      process.nextTick(emitErrorNT, this, exception);
+      this._writableState.errorEmitted = true;
     }
   }
 
   if (this.destroyed) {
     debug('already destroyed, fire error callbacks');
-    fireErrorCallbacks();
+    fireErrorCallbacks.call(this);
```

Comment:
What about just passing `this` to the function and have `self` be the parameter name?